test/SolutionSubmitter.spec.ts

```diff
@@ -0,0 +1,131 @@
+const ERC20 = artifacts.require("ERC20");
+const GasToken = artifacts.require("GasToken");
+const MockContract = artifacts.require("MockContract");
+const SolutionSubmitter = artifacts.require("SolutionSubmitter");
+
+import truffleAssert from "truffle-assertions";
+import {
+  closeAuction,
+  makeDeposits,
+  placeOrders,
+  setupGenericStableX,
+} from "./utilities";
+import { solutionSubmissionParams, basicRingTrade } from "./resources/examples";
+
+const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000";
+
+contract("SolutionSubmitter", (accounts) => {
+  const [owner, non_owner] = accounts;
+  describe("execute", () => {
+    it("allows arbitrary functions calls on behalf of instance from owner", async function () {
+      const instance = await SolutionSubmitter.new(
+        ZERO_ADDRESS,
+        ZERO_ADDRESS,
+        0,
+        { from: owner },
+      );
+
+      const mock = await MockContract.new();
+      const mockedToken = await ERC20.at(mock.address);
+      const transfer = mockedToken.contract.methods
+        .transfer(owner, 100)
+        .encodeABI();
+
+      await instance.execute(mock.address, transfer, { from: owner });
+
+      const invocationCount = await mock.invocationCountForMethod.call(
+        transfer,
+      );
+      assert.equal(1, invocationCount.toNumber());
+    });
+    it("reverts functions calls from non-owner", async function () {
```

Comment:
ubernit: Could you add whitespace between the tests? It makes it a lot harder for me personally to parse and separate unit tests at a quick glance.
```suggestion

    it("reverts functions calls from non-owner", async function () {
```